examples/shell/esp32/main/main.cpp

```diff
@@ -25,14 +25,42 @@
 #include "linenoise/linenoise.h"
 #include "nvs_flash.h"
 #include "support/CHIPMem.h"
+#include "support/ErrorStr.h"
 
 #include <ChipShellCollection.h>
 #include <lib/shell/shell_core.h>
 #include <lib/support/CHIPPlatformMemory.h>
 #include <platform/CHIPDeviceLayer.h>
 
+struct ShellLineArgs
+{
+    char * line;
+    TaskHandle_t source_task;
+};
+
+static void process_shell_line(intptr_t context)
+{
+    ShellLineArgs * args = reinterpret_cast<ShellLineArgs *>(context);
```

Comment:
can we somehow make this typesafe? 

Could we make ShellLineArgs have a NotifyDone/WaitForDone method to not have the hardcoded xTaskNotify inside the members? I would find that easier to read.

Reply:
The reinterpret cast is a must here.